### PR TITLE
[GNA plugin] Fix issues from static code analysis

### PR DESCRIPTION
**Pull request by rjeziers**
### Details:
 - This PR fixes issues in the GNA plugin discovered by static code analysis

### Tickets:
 - 93308

### CODE REVIEW

**Comment by marcinkusm** [comment]
changes looks fine.

**Review by marcinkusm — APPROVED** [review]
looks fine

File: src/plugins/intel_gna/src/backend/am_intel_dnn.cpp [hunk 1]
```diff
@@ -751,9 +751,11 @@ void GNAPluginNS::backend::AMIntelDNN::PrintOffset(std::ofstream& out, const std
         typeOfRegion = GNAPluginNS::memory::rRegionToStr(queue->regionType());
         offset = queue->getOffset(ptr).second;
     }
+    std::ios_base::fmtflags out_flags(out.flags());
     out << "<memory_region_type> " << typeOfRegion << "\n";
     out << "<" << type << "_address> "
         << "0x" << std::setfill('0') << std::setw(8) << std::hex << offset << "\n";
+    out.flags(out_flags);
```
Comment (mryzhov) [inline src/plugins/intel_gna/src/backend/am_intel_dnn.cpp:758]: It is not necessary to reset flags manually

File: src/plugins/intel_gna/src/layers/gna_layer_info.hpp [hunk 2]
```diff
@@ -62,15 +62,15 @@ class LayerInfo {
     // The name of the funciton may be somehwat misleading
     // Explanation: when in low precision mode the listed layers have 8-bit outputs
     // and when in 16-bit input mode, they have 16-bit outputs
-    bool has8BOr16BOutput() const noexcept {
```
Comment (mryzhov) [inline src/plugins/intel_gna/src/layers/gna_layer_info.hpp:65]: Are you sure that the right fix to allow throw exceptions in this function? 
IMO this function should return true or false and should not throw any exceptions.

Comment (marcinkusm) [reply src/plugins/intel_gna/src/layers/gna_layer_info.hpp:65]: I guess this is necessery if don't want to change rest of the code
there is used isCropAffined() which calls GetCropParams() which calls get_crop_params)_ this is may throw.
if we don't want to have excetion in this methods we need to redesign this methods to return error information as argument and handle this one level upper.

currently this method can throw but noexcept suggest the it shouldn't.

Comment (mryzhov) [reply src/plugins/intel_gna/src/layers/gna_layer_info.hpp:65]: I'm ok with that solution, because this code is legacy.

**Review by mryzhov — APPROVED** [review]

**Pull request merged**